packages/core/src/local-workspace/remote_map/remote_map.ts

```diff
@@ -566,7 +566,7 @@ export const createRemoteMapCreator = (
       value,
     }: remoteMap.RemoteMapEntry<string>): Promise<remoteMap.RemoteMapEntry<T, K>> => {
       const cacheKey = keyToTempDBKey(key)
-      const cacheValue = locationCache.get(cacheKey) as T | undefined
+      const cacheValue = await (locationCache.get(cacheKey) as Promise<T | undefined>)
```

Comment:
this cast is incorrect, the type here is actually `Promise<T | undefined> | undefined` (the second `undefined` is because it is possible that the `cacheKey` is not in the cache...

it might be better to leave the cast as it was and put it outside the `await`
```suggestion
      const cacheValue = (await locationCache.get(cacheKey)) as T | undefined
```